### ISSUE TITLE
Problem with refresh (after resize or rotation)

### ISSUE THREAD

**Issue by skrew** [kean/NukeUI#14]
Hi,

I'm using 2 types of images (one for compact, one for regular) and i'm switching images after a device rotation or window resize.

But the good image is loaded only the first time, after a device rotation, the result are "inversed", ie: seeing portrait image in landscape and landscape image in portrait.

You can test with this code on an iPhone and rotate the device:

```
import SwiftUI
import NukeUI


struct TestView: View {
    
    @Environment(\.horizontalSizeClass) var wSizeClass
    @Environment(\.verticalSizeClass) var hSizeClass
    
    var body: some View {
        VStack {
            Text("LazyImage").foregroundColor(.white)
            LazyImage(source: hSizeClass == .regular ? "https://liquidchaostattoos.com/wp-content/uploads/2015/06/artist-portrait-placeholder.jpg" : "https://www.ctsfw.edu/wp-content/uploads/2016/02/landscape-placeholder-image.jpg")
                .frame(width: 150, height: 150)
            
            
            Text("AsyncImage").foregroundColor(.white)
            if #available(iOS 15.0, *) {
                AsyncImage(url: hSizeClass == .regular ?
                           URL(string: "https://liquidchaostattoos.com/wp-content/uploads/2015/06/artist-portrait-placeholder.jpg")! :
                            URL(string: "https://www.ctsfw.edu/wp-content/uploads/2016/02/landscape-placeholder-image.jpg")!) { image in
                    image.resizable()
                } placeholder: {
                    ProgressView()
                }
                .frame(width: 150, height: 150)
            } else {
                // Fallback on earlier versions
            }
        }
    }
}

struct TestView_Previews: PreviewProvider {
    static var previews: some View {
        NavigationView {
            TestView()
                .environment(\.colorScheme, .dark)
        }
    }
}
```

**Comment by kean** [comment]
Hi, the "change the source" scenario should be working as expected. `LazyImage` does it using [onChange(of:)](https://developer.apple.com/documentation/swiftui/view/onchange(of:perform:)). Are you sure you are setting the correct URL based on the size class?

**Comment by skrew** [comment]
Yes, sure. I have edited the code in my previous post to include the new `AsyncImage` from `iOS 15`.

At first launch, portrait mode, both have the same (good) image:
![Simulator Screen Shot - iPhone 11 - 2021-07-22 at 23 38 22](https://user-images.githubusercontent.com/738170/126713028-226da37e-5841-462e-94d9-8ca6b92392c1.png)

After the rotation to landscape, you can see `LazyImage` still display the portrait one
![Simulator Screen Shot - iPhone 11 - 2021-07-22 at 23 38 30](https://user-images.githubusercontent.com/738170/126713088-0b7a829f-7905-4c13-8b62-d308f536d06f.png)

And after back to portrait, `LazyImage` display the landscape one... `AsyncImage` still display the good one
![Simulator Screen Shot - iPhone 11 - 2021-07-22 at 23 38 35](https://user-images.githubusercontent.com/738170/126713143-af7943ea-01ea-4521-9802-1b45326052e9.png)

**Comment by kean** [comment]
Hey, @skrew. Should be fixed in [0.6.5](https://github.com/kean/NukeUI/releases/tag/0.6.5). I misused `onChange(of:)`, it's such a PITA.

**Comment by skrew** [comment]
It works fine, thanks ! :)
